ecommerce/crm/lib/crm/order.rb

```diff
@@ -1,14 +1,13 @@
 module Crm
   class Order
     include AggregateRoot
-    CustomerAlreadyAssigned = Class.new(StandardError)
 
     def initialize(id)
       @id = id
     end
 
     def set_customer(customer_id)
-      raise CustomerAlreadyAssigned if @customer_id
+      return if @customer_id == customer_id
```

Comment:
I think we need to allow users to reassign customers. If order submission fails the first time the user is still being assigned. We then display the form with an error again and the user can change the customer.